src/hooks/core/websocket/useBetterWebsocketHook.ts

```diff
@@ -36,18 +38,25 @@ const useBetterWebsocketHook = () => {
     );
   };
 
-  const initAuthorization = async () => {
-    const token: string = TokenStorage.get(ITokenEnum.anonymousToken);
-    const userId: string = await getAnonymousUserId();
+  const initAuthorization = async (isAnonymous: boolean) => {
+    const token =
+      TokenStorage.get(isAnonymous ? ITokenEnum.anonymousToken : ITokenEnum.token) ?? '';
+    const userId = isAnonymous ? await getAnonymousUserId() : await getUserId();
+
+    const urlEncodedData = generateUserDataUrlEncoded({
+      userId,
+      token,
+      isAnonymous
+    });
 
-    const urlEncodedData = generateUserDataUrlEncoded(userId, token);
     const websocketUrl = generateWebsocketUrl(urlEncodedData, token);
     return websocketUrl;
   };
 
-  const {lastJsonMessage} = useSimpleWebsocketHooks(initAuthorization());
+  const {lastJsonMessage: lastAnonymousMessage} = useSimpleWebsocketHooks(initAuthorization(true));
+  const {lastJsonMessage: lastSignedMessage} = useSimpleWebsocketHooks(initAuthorization(false));
 
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initAuthorization` function has been updated to accept a boolean parameter `isAnonymous`. This change makes the function more versatile as it can now initialize authorization for both anonymous and signed-in users. The function now returns a websocket URL based on whether the user is anonymous or not. However, similar to the previous comment, there's no null check for the `token` and `userId`. If either of these is `null` or `undefined`, it could lead to unexpected behavior.
```diff
-   const token =
-     TokenStorage.get(isAnonymous ? ITokenEnum.anonymousToken : ITokenEnum.token) ?? '';
-   const userId = isAnonymous ? await getAnonymousUserId() : await getUserId();
+   const token =
+     TokenStorage.get(isAnonymous ? ITokenEnum.anonymousToken : ITokenEnum.token);
+   if (!token) {
+     throw new Error('Token is not available');
+   }
+   const userId = isAnonymous ? await getAnonymousUserId() : await getUserId();
+   if (!userId) {
+     throw new Error('User ID is not available');
+   }
```
The `useBetterWebsocketHook` function now returns two messages: `lastAnonymousMessage` and `lastSignedMessage`. This change allows the hook to handle both anonymous and signed-in users, which improves the versatility of the code.


<!-- This is an auto-generated comment by OSS CodeRabbit -->